src/audio/pipeline/pipeline-stream.c

```diff
@@ -60,10 +71,6 @@ pipeline_should_report_enodata_on_trigger(struct comp_dev *rsrc,
 	if (pipe_source && dev_comp_type(pipe_source) == SOF_COMP_DAI)
 		return false;
 
-	/* source pipeline may not be active since priority is not higher than current one */
-	if (rsrc->pipeline->priority <= ppl_data->start->pipeline->priority)
-		return false;
```

Comment:
is this intended or a left-over?

Reply:
seems to be a bad hunk

Reply:
A issue found on windows

Reply:
This is intended @lyakh . @RanderWang  If IPC4 build, we will return "false" already at the beginning of the function, so this should fix the issue you saw on Windows.

Reply:
In case of IPC3, the remove check to compare priorities seems wrong. IPC3 has not separate trigger order, so the check should actually be reversd for IPC3. If "rsrc" has higher priotiy, then we cannot check it state yet. In practise IPC3 pipelines have priority set to 0 so this is not hit, but I think it's better to remove.

Reply:
@lyakh  Ok let me remove this. I think the hunk was added to solve a IPC4 specific problem (but didn't cause harm in IPC3 builds), but I admit it's very nonobvious toconclude why this code can be removed. It apparently does no harm for IPC3, so let's just limit the complexity of the patch and simply not touch this code.

Reply:
@lyakh bad hunk removed